front/src/hooks/useQrResolver.ts

```diff
@@ -56,68 +56,64 @@ export const useQrResolver = () => {
           fetchPolicy,
         })
         .then(({ data, errors }) => {
-          if ((errors?.length || 0) > 0) {
-            const errorCode = errors ? errors[0]?.extensions?.code : undefined;
-            if (errorCode === "FORBIDDEN") {
-              triggerError({
-                message: "You don't have permission to access this box!",
-              });
-              return {
-                kind: IQrResolverResultKind.NOT_AUTHORIZED,
-                qrHash: hash,
-              } as IQrResolvedValue;
-            }
-            if (errorCode === "BAD_USER_INPUT") {
-              triggerError({
-                message: "No box found for this QR code!",
-              });
-              return {
-                kind: IQrResolverResultKind.NOT_FOUND,
-                qrHash: hash,
-              } as IQrResolvedValue;
-            }
+          if (errors?.length) {
+            // Likely an unexpected graphQL error.
             triggerError({
               message: "QR code lookup failed. Please wait a bit and try again.",
             });
+
             return {
               kind: IQrResolverResultKind.FAIL,
               qrHash: hash,
             } as IQrResolvedValue;
           }
-          // @ts-ignore
-          if (!data?.qrCode?.box) {
+
+          if (data.qrCode.__typename === "ResourceDoesNotExistError") {
             return {
               kind: IQrResolverResultKind.NOT_ASSIGNED_TO_BOX,
```

Comment:
I think this should be `NO_BOXTRIBUTE_QR` and also show an error toast?

Reply:
This is handled at `/front/src/components/QrReader/QrReaderContainer.tsx`

```ts
case IQrResolverResultKind.NOT_ASSIGNED_TO_BOX: {
  if (!multiScan) {
    onSuccess();
    navigate(`/bases/${baseId}/boxes/create/${qrResolvedValue?.qrHash}`);
  } else {
    triggerError({
      message: "No box associated to this QR code!",
    });
    setIsProcessingQrCodeDelayed(false);
  }
  break;
}
```

Reply:
yes it works :+1: But I end up on an empty page `http://localhost:3000/bases/1/boxes/create/13f12820c8010f2f7349962930e6bf4` (I used the last one of the invalid QR codes from the repo readme).
I'd expect to stay on `/qrread`

Reply:
Huh, this should've been handled at `front/src/hooks/useQrResolver.ts`

```ts
const resolveQrCode = useCallback(
  async (qrCodeUrl: string, fetchPolicy: FetchPolicy): Promise<IQrResolvedValue> => {
    setLoading(true);
    const extractedQrHashFromUrl = extractQrCodeFromUrl(qrCodeUrl);

    if (!extractedQrHashFromUrl) {
      triggerError({
        message: "This is not a Boxtribute QR code!",
      });

      setLoading(false);
      return { kind: IQrResolverResultKind.NOT_BOXTRIBUTE_QR } as IQrResolvedValue;
    }

    const qrResolvedValue: IQrResolvedValue = await resolveQrHash(
      extractedQrHashFromUrl,
      fetchPolicy,
    );

    setLoading(false);
    return qrResolvedValue;
  },
  [resolveQrHash, triggerError],
);
```

Cheking this out.